wd_helper.au3

```diff
@@ -1187,6 +1187,104 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
+	Local $iErr = $_WD_ERROR_Success
+	Local $sStartLocation = '', $sMessage = ''
+
+	Local $aFrameList = _WD_FrameList($sSession, True)
+	$iErr = @error
+	If $iErr Then
+		$iErr = $_WD_ERROR_GeneralError
+		$sMessage = ' > Issue with getting list of frames'
+	Else
+		Local $iFrameCount = UBound($aFrameList, $UBOUND_ROWS)
+
+		For $i = 0 To $iFrameCount - 1
+			If $aFrameList[$i][$_WD_FRAMELIST_Relative] = '' Then $sStartLocation = $aFrameList[$i][$_WD_FRAMELIST_Absolute]
+		Next
+
+		#Region ; this region is prevented from redundant logging ( _WD_FrameEnter and _WD_FindElement ) if not in Full debug mode > https://github.com/Danp2/au3WebDriver/pull/290#issuecomment-1100707095
+		Local $_WD_DEBUG_Saved = $_WD_DEBUG ; save current DEBUG level
+		If $_WD_DEBUG <> $_WD_DEBUG_Full Then $_WD_DEBUG = $_WD_DEBUG_None
+
+		For $i = $iFrameCount - 1 To 0 Step -1
+			_WD_FrameEnter($sSession, $aFrameList[$i][$_WD_FRAMELIST_Absolute])
+			$iErr = @error
+			If $iErr Then
+				If Not $_WD_DetailedErrors Then $iErr = $_WD_ERROR_Exception
+				$sMessage = ' > Issue with entering frame=' & $aFrameList[$i][$_WD_FRAMELIST_Absolute] & '  URL=' & $aFrameList[$i][$_WD_FRAMELIST_URL]
+				ExitLoop
+			Else
+				$aFrameList[$i][$_WD_FRAMELIST_MatchedElements] = _WD_FindElement($sSession, $sStrategy, $sSelector, Default, True, $bShadowRoot)
+				$iErr = @error
+				If $iErr = $_WD_ERROR_Success Then
+					ContinueLoop ; keep the frame in the list and continue searching in next frame
+				ElseIf $iErr = $_WD_ERROR_NoMatch Then ; element was not found on location: ' & $aFrameList[$i][$_WD_FRAMELIST_Absolute]
+					_ArrayDelete($aFrameList, $i) ; delete frame from the list because the searched element do not exist within the frame
+					ContinueLoop
+				Else
+					If Not $_WD_DetailedErrors Then $iErr = $_WD_ERROR_Exception
+					$sMessage = ' > Issue with finding element in frame=' & $aFrameList[$i][$_WD_FRAMELIST_Absolute] & '  URL=' & $aFrameList[$i][$_WD_FRAMELIST_URL]
+					$aFrameList[$i][$_WD_FRAMELIST_MatchedElements] = ''
+					ExitLoop
+				EndIf
+			EndIf
+		Next
+
+		If $i = -1 Then ; all frames was checked
+			If UBound($aFrameList) Then
+				$iErr = $_WD_ERROR_Success
+			Else
+				$iErr = $_WD_ERROR_NoMatch
+			EndIf
+		EndIf
+
+		If $sStartLocation Then ; Back to "calling frame"
+			_WD_FrameEnter($sSession, $sStartLocation)
+			$iErr = @error
+			If $iErr And Not $_WD_DetailedErrors Then $iErr = $_WD_ERROR_Exception
+			#TODO @error REFACTORING NEEDED HERE
+		EndIf
+
+		$_WD_DEBUG = $_WD_DEBUG_Saved ; restore DEBUG level
+		$sMessage = $sParameters & $sMessage
+		#EndRegion ; this region is prevented from redundant logging ( _WD_FrameEnter and _WD_FindElement ) if not in Full debug mode > https://github.com/Danp2/au3WebDriver/pull/290#issuecomment-1100707095
+
+		If $sStartLocation = '' Or ($iErr And $iErr <> $_WD_ERROR_NoMatch) Then
+			$sMessage = ' > Was not able to check / back to "calling frame" : StartLocation=' & $sStartLocation
+		EndIf
```

Comment:
I have some questions about this -

- Couldn't this be move up to be an `Else` to `If` on line 1267?
- Doesn't this overwrite any prior text assigned to `$sMessage`?

Reply:
> * Doesn't this overwrite any prior text assigned to `$sMessage`?

Yes. Fixed: [ampersand](https://github.com/Danp2/au3WebDriver/pull/433/commits/f7844c42710283cc0dcd0850911a8634031365df)

Reply:
> * Couldn't this be move up to be an `Else` to `If` on line 1267?

No.
Take a look on: 
https://github.com/Danp2/au3WebDriver/blob/85b133ca755a26f49d9e14d0784161e7913bec94/wd_helper.au3#L1267-L1272
What will happen when this part will set error ?

Reply:
btw. [#TODO removed](https://github.com/Danp2/au3WebDriver/pull/433/commits/555479f8ec42e70692c6de8c9be15d4699145041)

Reply:
You handle this a little differently in _WD_FrameList.

Reply:
Good point. Take a look on: [Back to "calling frame" - $sMessage reporting](https://github.com/Danp2/au3WebDriver/pull/433/commits/934b0f87926b71a35db8797aa7e9cf4963da6562)

Reply:
What about the `Else` portion that could trigger if $sStartLocation wasn't set?

Reply:
If it will not set ? you mean in `_WD_FrameListFindElement` ?
It is checked after `_WD_FrameList` was called, so if `_WD_FrameList` was not set error this mean `$sStartLocation` is able to be checked and set.

Reply:
> so if `_WD_FrameList` as so far not set error this mean `$sStartLocation` is able to be checked and set.

I mean `_WD_FrameList` internally do the same - check if `$sStartLocation` is able to be checked and set.

Reply:
Yes, I see that now. 🙂